### ISSUE TITLE
Using in a render() method?

### ISSUE THREAD

**Issue by mikelambert** [taggon/react-native-intl#1]
So I understand I can construct a translator object and use _ in my render() method without any problems. But I'm confused how I would go about using date/time formatting functions within a render() method, due to their asynchronous nature. I have a proposed solution below, but I believe this results in a flash of "empty data" before showing the real data (similar to the flash-of-unstyled-text in the HTML world), and am curious if there's a better way...a pure JS solution, a la one of the other libraries out there. Thanks!

```
class TranslatedDate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {translatedDate: null};
    this.translate();
  }

  async translate() {
    const translatedDate = await new Intl.DateTimeFormat('en-US').format(date);
    this.setState({translatedDate});
  }

  render() {
    return <Text>{this.state.translatedDate}</Text>;
  }
}

```

**Comment by taggon** [comment]
I think your code will work if you fix these errors:
- `this.setState(translatedDate);` should be `this.setState( {translatedDate} );` or `this.setState( {'translatedDate': translatedDate} )`;
- Missing `}` after `{this.state.translatedDate`

And I already showed how to use `DateTimeFormat` and `NumberFormat` object here:
https://github.com/taggon/react-native-intl/blob/master/Examples/Hello/index.ios.js

Let me know if you have any further questions. :)

**Comment by mikelambert** [comment]
Sorry, it was written directly without any attempt at compiling. Fixing the syntax errors, I'm sure the code will work.

But as stated in my original message, `I believe this results in a flash of "empty data" before showing the real data`. The first frame (or two) will show an empty <Text> before it loads the formatted-date back from the async call and fills out the <Text> with the actual date.

In other words, this library's reliance on native date-time processing (which other i18n/l11n for RN libraries don't use) means it must operate with an async/promise API, and means it's impossible to get a formatted date in the initial render() call. Is this correct? (And perhaps it's worth listing that tradeoff in the docs)

**Comment by francisrod01** [comment]
So.. @mikelambert There is a problem to render in a `ListItem` of `react-native-elements` because the element in a list invokes this async method like this:

parse-times.js

```js
const formatNumber = async (number, hasSymbol) => {
    const symbol = (hasSymbol) ? numberOpts : {}

    return await new Intl.NumberFormat(parseLocaleUser, symbol).format(number)
}
```

list-data.js

```js
async formatNumber(number) {
    const formattedNumber = await parseTimes.formatNumber(number, true)
    
    this.setState({ formattedNumber })

    console.log('========== parse number: ', this.state.formattedNumber)
}

..

<Card
    title={"Results for this tags: #motorboat, #ride, #boat"}
>
    { services.map((s, i) => {
        this.formatNumber(s.price)

        return (
            <ListItem
                key={i}
                leftIcon={{
                    name: 'circle-o',
                    type: 'font-awesome',
                    style: { color: 'gray' },
                }}
                title={
                    <Text style={styles.medium}>{ s.name }</Text>
                }
                subtitle={
                    <View style={styles.row}>
                        <View style={styles.column}>
                            <Text style={{ color: 'steelblue' }}>
                                { this.showAnothers(s) }
                            </Text>
                        </View>
                    </View>
                }
                badge={{
                    value: this.state.formattedNumber,
                    textStyle: { color: 'orange' },
                    containerStyle: { marginTop: 30 },
                }}
                onPress={() => this.goToDetails(s)}
            />
        )
    })}
</Card>
```

It works but `formatNumber()` method is always called, and this values are constantly changed in the screen.

**Comment by batistafrancis** [comment]
The same problem here: https://github.com/taggon/react-native-intl/issues/8
